### PR TITLE
Register lookups for all defined languages

### PR DESCRIPTION
**Pull request by madig**
Closes https://github.com/googlefonts/ufo2ft/issues/699.

I'm not sure about:

1. Should we also consider languages defined for `DFLT`? The spec technically allows this but I don't know if it's a valid use case?
2. If the designer defined a language for e.g. the OT tag `tml2`, should it also be inserted for `taml`?

### CODE REVIEW

**Comment by anthrotype** [comment]
> Should we also consider languages defined for DFLT?

yes, I think so, if they are expliclity defined, I don't see why we should exclude those

> If the designer defined a language for e.g. the OT tag tml2, should it also be inserted for taml?

no, we should only do what being told

File: Lib/ufo2ft/featureWriters/kernFeatureWriter.py [hunk 1]
```diff
@@ -198,6 +198,19 @@ def setContext(self, font, feaFile, compiler=None):
         ctx.gdefClasses = self.getGDEFGlyphClasses()
         ctx.glyphSet = self.getOrderedGlyphSet()
 
+        # Remember which languages are defined for which OT tag, as all
+        # generated kerning needs to be registered for the script's `dflt`
+        # language, but also all those the designer defined manually. Otherwise,
+        # setting a different language for a script would deactivate kerning.
+        # NOTE: This does not contain languages defined for DFLT. The spec
+        # technically allows it, but
```
Comment (anthrotype) [inline Lib/ufo2ft/featureWriters/kernFeatureWriter.py:206]: let's change `ast.getScriptLanguageSystems` to optionally not skip DFLT

**Comment by madig** [comment]
Validated these changes with https://github.com/daltonmaag/kerning-validator/commit/16927fc10b1f4d83d0f2b029b4c1b16a8d5c7efe.

File: Lib/ufo2ft/featureWriters/kernFeatureWriter.py [hunk 2]
```diff
@@ -198,6 +198,17 @@ def setContext(self, font, feaFile, compiler=None):
         ctx.gdefClasses = self.getGDEFGlyphClasses()
         ctx.glyphSet = self.getOrderedGlyphSet()
 
+        # Remember which languages are defined for which OT tag, as all
+        # generated kerning needs to be registered for the script's `dflt`
+        # language, but also all those the designer defined manually. Otherwise,
+        # setting any language for a script would deactivate kerning.
+        feaScripts = ast.getScriptLanguageSystems(feaFile, excludeDflt=False)
```
Comment (anthrotype) [inline Lib/ufo2ft/featureWriters/kernFeatureWriter.py:205]: nit: I would call this feaLanguages or feaLangsByScript -- the scripts are the key, but the values are lists of languages, the values is the important bit usually

**Review by anthrotype — APPROVED** [review]
LGTM with minor comments

**Pull request merged**